### DISCUSSION TITLE
importing dataset with .its that need to be anonymized

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#192] [Q&A] [answered]
I'm working on the Rague dataset, and so far I've done:

```
#steps taken to create this
source ~/ChildProjectVenv/bin/activate

datalad create rague

cd rague
mkdir scripts
mkdir metadata
mkdir -p recordings/raw
mkdir extra
touch extra/.gitignore # Make sure the directory is present even though it's empty
mkdir -p annotations/its/raw
mkdir annotations/its/confidential/

# move annotations
cp /Users/acristia/Box/data_broad_strokes_lena/rague/ITS_FILES_OSF_3.15.2019/*.its annotations/its/confidential/.

# working on the metadata
# I took metadata.py from bergelson (within EL1000)
cp /Users/acristia/Box/data_broad_strokes_lena/rague/purdue_ndd_readme.txt extra/.
mkdir -p metadata/confidential/original/
cp /Users/acristia/Box/data_broad_strokes_lena/rague/purdue_ndd_metadata.csv metadata/confidential/original/.
python scripts/metadata.py rague


child-project validate . --ignore-files

echo "* annex.backend=MD5E
**/.git* annex.largefiles=nothing
* annex.largefiles=nothing" > .gitattributes


datalad save . -m "first commit"

```

Notice I've done a first local commit, which has the its in annotations/its/raw because I thought that was what was needed given how metadata.py was set up

However, now I'm running import_its.py (taken from kidd) and I get this:

```
python scripts/import_its.py
```

> /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame
See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  errors=errors,
scripts/import_its.py:14: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead
See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  input['filter'] = input['recording_filename'].str.extract(r"_([0-9])$")
could not delete './annotations/its/converted', as it does not exist (yet?)

I see no files in confidential/, all the files that were there before in raw/ and some files in converted/

I think what I should have done is:

- make sure that .gitignore contains the patterns for confidential
- also double check with you .gitattributes, which is probably improperly specified right now
- modify metadata.py so that it assumes .its files that have not yet been anonymized will be in confidential
- modify import_its.py so that it assumes .its files that have not yet been anonymized will be in confidential

did I get all of that right?
I really don't mind starting over from scratch to make sure that the history of the datalad dataset is clean

**Comment by lucasgautheron** [comment, marked as answer]
Honestly you did better than I would have expected knowing that I left no instruction whatsoever about how to migrate datasets!

These are the steps that I think should work after you have created the structure and imported the original metadata as you did:

 - Make sure you have copied the its to to `annotations/its/confidential/raw`if the originals are confidential, or straight to `annotations/its/raw/` if they aren't.
 - copy the scripts in https://gin.g-node.org/EL1000/bergelson/src/main/scripts to your dataset `scripts/` folder.
 - Run the metadata conversion script: `python scripts/metadata.py rague metadata/confidential/original/purdue_ndd_metadata.csv`
 - Anonymize the its if the originals are confidential (`child-project anonymize . --input-set its/confidential --output-set its`)
 - Import the its: `python scripts/import.py its`

**Reply by alecristia** [reply]
and you confirm that this is enough:

```
echo "* annex.backend=MD5E
**/.git* annex.largefiles=nothing
scripts/* annex.largefiles=nothing
metadata/*.csv annex.largefiles=nothing
recordings/* annex.largefiles=((mimeencoding=binary)and(largerthan=0))" > .gitattributes
```

because permissions regarding confidential is handled at the package level, I presume?

**Reply by lucasgautheron** [reply]
To create EL1000 datasets, I recommend that you follow these instructions rather than doing it from scratch: https://github.com/LAAC-LSCP/datalad-procedures

The EL1000 procedure structures your dataset with the proper format and the right `.gitattributes` values.
It will also configure gin siblings automatically.

Once you have created the dataset according to the instructions in the README of datalad-procedures, you can add the its, import the metadata, etc.

This should be easier (hopefully!)

**Reply by alecristia** [reply]
ah got it! I was just looking into that
